core/drivers/pl022_spi.c

```diff
@@ -181,13 +181,21 @@ enum pl022_spi_mode {
 	PL022_SPI_MODE3 = SSPCR0_SPO1 | SSPCR0_SPH1  /* 0xC0 */
 };
 
+/*
+ * @param[inout]	*num_rxpkts	number of max packets to receive
+ */
 static void pl022_txrx8(struct spi_chip *chip, uint8_t *wdat,
 	uint8_t *rdat, size_t num_txpkts, size_t *num_rxpkts)
 {
 	size_t i = 0;
 	size_t j = 0;
 	struct pl022_data *pd = container_of(chip, struct pl022_data, chip);
 
+	if (*num_rxpkts != num_txpkts) {
+		MSG("WARNING: *num_rxpkts != num_txpkts!\n");
+		MSG("WARNING: Is this what you really want?\n");
+	}
+
```

Comment:
Well, if I'm not mistaken the whole point of this PR was to document `num_rxpkts` and perform some validations on it. Now you're saying that `num_rxpkts` doesn't make sense at all, which I'm perfectly fine with (the simpler the better! and thanks for the links BTW). So I think we should just drop this PR and create a new one which simply removes `num_rxpkts` (and possibly renames `num_txpkts` to `num_pkts`).
What do you think?